string mode
bool bar
bool traffic_light
